echopype/mask/api.py

```diff
@@ -17,6 +17,231 @@
+        single input or list that corresponds to a DataArray or a path to a DataArray. If a path
+        is provided this should point to a zarr or netcdf file with only one array in it.
+    storage_options_mask: dict or list of dict, default={}
+        Any additional parameters for the storage backend, corresponding to the
+        path provided for ``mask``. If ``mask`` is a list, then this input should either
+        be a list of dictionaries or a single dictionary with storage options that
+        correspond to all elements in ``mask`` that are paths.
+
+    Returns
+    -------
+    xr.DataArray or list of xr.DataArray
+        If the ``mask`` input is a single value, then the corresponding DataArray will be
+        returned, else a list of DataArrays corresponding to the input masks will be returned
+
+    Raises
+    ------
+    RuntimeError
+        If ``mask`` is a single-element and ``storage_options_mask`` is not a single dict
+    TypeError
+        If ``storage_options_mask`` is not a list of dict or a dict
+    """
+
+    if isinstance(mask, list):
+
+        # if storage_options_mask is not a list create a list of
+        # length len(mask) with elements storage_options_mask
+        if not isinstance(storage_options_mask, list):
+
+            if not isinstance(storage_options_mask, dict):
+                raise TypeError("storage_options_mask must be a list of dict or a dict!")
+
+            storage_options_mask = [storage_options_mask] * len(mask)
+        else:
+            # ensure all element of storage_options_mask are a dict
+            if not all([isinstance(elem, dict) for elem in storage_options_mask]):
+                raise TypeError("storage_options_mask must be a list of dict or a dict!")
+
+        for mask_ind in range(len(mask)):
+
+            # validate the mask type or path (if it is provided)
+            mask_val, file_type = validate_source_ds_da(
+                mask[mask_ind], storage_options_mask[mask_ind]
+            )
+
+            # replace mask element path with its corresponding DataArray
+            if isinstance(mask_val, str):
+                # open up DataArray using mask path
+                mask[mask_ind] = xr.open_dataarray(
+                    mask_val, engine=file_type, chunks="auto", **storage_options_mask[mask_ind]
+                )
+
+    else:
+
+        if not isinstance(storage_options_mask, dict):
+            raise RuntimeError(
+                "The provided input storage_options_mask should be a single "
+                "dict because mask is a single value!"
+            )
+
+        # validate the mask type or path (if it is provided)
+        mask, file_type = validate_source_ds_da(mask, storage_options_mask)
+
+        if isinstance(mask, str):
+            # open up DataArray using mask path
+            mask = xr.open_dataarray(mask, engine=file_type, chunks="auto", **storage_options_mask)
```

Comment:
While working on another PR I found that `chunks="auto"` will fail for netcdf. This should be replaced with `chunks={}`.